graders/python/python_autograder/pl_main.py

```diff
@@ -115,6 +116,13 @@ def add_files(results):
         if len(format_errors) > 0:
             grading_result["format_errors"] = format_errors
 
+        # Instructors may have named their tests incorrectly or somehow misconfigured things.
+        # Help point them in the right direction.
+        if len(results) == 0:
+            grading_result["message"] = (
+                "No tests were found. Please check your test cases."
```

Comment:
I don't think this should be an error. Cases like compilation errors often happen before tests are created, and in that case the message and/or output will have relevant content without any tests present. I think a generic message is fine, but only if both message and output are blank, and should not suggest an action. Something like "No tests were found" is enough.

Reply:
I just tested the autograder with the changes above. If I submit an answer with compilation errors, the error message above is added, even though the question has tests. The tests were not reached because of the compilation error, so this message was actually incorrect/misleading.

Upon further review of this code, I believe there are better ways to identify if a problem is indeed a case where no tests were found, such as checking if `gradable` is `True` or keeping track of the `TestLoader` results in line 63, calling [`countTestCases()`](https://docs.python.org/3/library/unittest.html#unittest.TestSuite.countTestCases).

Reply:
I changed to using `suite.countTestCases() > 0` in aaeaae7. How does that look?